docs/design/values.md

```diff
@@ -858,6 +868,32 @@ types in Carbon provide access to an object with an API subset that can help
 model important requirements like ensuring usage is exclusively by way of a
 _thread-safe_ interface subset of an otherwise _thread-compatible_ type.
 
+Note that `const T` is a type qualification and is generally orthogonal to
+expression categories or what form of pattern is used, including for object
+parameters. Notionally, it can occur both with `addr` and value object
+parameters. However, on value patterns, it is redundant as there is no
+meaningful distinction between a value expression of type `T` and type
+`const T`. For example, given a type and methods:
+
+```carbon
+class X {
+  fn Method[self: Self]();
+  fn ConstMethod[self: const Self]();
+  fn AddrMethod[addr self: Self*]();
+  fn AddrConstMethod[addr self: const Self*]();
+}
+```
+
+The methods can be called on different kinds of expressions according to the
+following table:
+
+|  Expression category: | `let x: X` (value) | `let x: const X` (const value) | `var x: X` (reference) | `var x: const X` (const reference) |
```

Comment:
Optional: I think this might look a bit better with a forced newline

```suggestion
|  Expression category: | `let x: X`<br>(value) | `let x: const X`<br>(const value) | `var x: X`<br>(reference) | `var x: const X`<br>(const reference) |
```